### PR TITLE
The Most Non-Edgey PR Ever

### PR DESCRIPTION
**Pull request by Fox-McCloud**
Removes the `edge` var.

This was pretty much redundant; having `sharp` and `edge` both was excessive; there was quite literally 4 objects in the game that had `sharp` set to `1`, but `edge` set to `0`; two of them were bullet casings (one of which self deletes after firing).

The primary difference is that `sharp` controlled the type of damage it would doe (sharp vs blunt), and `edge` controlled if it would dismember or not.

Quite frankly, it's so edge case that you'd want something to cut but not dismember that it's not worth keeping this around or maintaining it, especially with such similar names.

- Completely guts the edge var
- Renames `DROPLIB_EDGE` to `DROPLIMB_SHARP`

Also I removed the sharp ***and*** edge vars from projectiles because projectiles don't use them *at all* and dismemberment of projectiles is handled by an entirely different variable.

### CODE REVIEW

File: code/game/objects/items/weapons/weaponry.dm [hunk 1]
```diff
@@ -83,7 +81,6 @@
 /obj/item/weapon/harpoon
```
Comment (Fox-McCloud) [inline code/game/objects/items/weapons/weaponry.dm]: Instance number 1 of sharp, but no edge.

File: code/modules/mob/living/living_defense.dm [hunk 2]
```diff
@@ -41,12 +41,6 @@
 /mob/living/bullet_act(var/obj/item/projectile/P, var/def_zone)
 	//Armor
 	var/armor = run_armor_check(def_zone, P.flag, armour_penetration = P.armour_penetration)
-	var/proj_sharp = is_sharp(P)
-	var/proj_edge = has_edge(P)
-	if((proj_sharp || proj_edge) && prob(getarmor(def_zone, P.flag)))
-		proj_sharp = 0
-		proj_edge = 0
```
Comment (Fox-McCloud) [inline code/modules/mob/living/living_defense.dm]: Let's do a bunch of math on if a projectile is sharp or has an edge, then never use it!

File: code/modules/mob/living/silicon/robot/component.dm [hunk 3]
```diff
@@ -34,13 +34,15 @@
 	installed = -1
 	uninstall()
 
-/datum/robot_component/proc/take_damage(brute, electronics, sharp, edge)
-	if(installed != 1) return
+/datum/robot_component/proc/take_damage(brute, electronics, sharp)
```
Comment (Fox-McCloud) [inline code/modules/mob/living/silicon/robot/component.dm]: this was clearly copy-pasted from  human organ damage at one point; the `sharp` var here isn't even used for ANYTHING.

File: code/modules/mob/living/simple_animal/parrot.dm [hunk 4]
```diff
@@ -477,7 +477,7 @@
 				var/mob/living/carbon/human/H = parrot_interest
 				var/obj/item/organ/external/affecting = H.get_organ(ran_zone(pick(parrot_dam_zone)))
 
-				H.apply_damage(damage, BRUTE, affecting, H.run_armor_check(affecting, "melee"), sharp=1)
+				H.apply_damage(damage, BRUTE, affecting, H.run_armor_check(affecting, "melee"), sharp = 1)
```
Comment (Fox-McCloud) [inline code/modules/mob/living/simple_animal/parrot.dm]: Whoah guys, parrots may be able to dismember now, srs buff.

File: code/modules/projectiles/guns/alien.dm [hunk 5]
```diff
@@ -49,8 +49,6 @@
 	name = "alloy spike"
 	desc = "A broadhead spike made out of a weird silvery metal."
 	projectile_type = /obj/item/projectile/bullet/spike
-	sharp = 1
-	edge = 0
```
Comment (Fox-McCloud) [inline code/modules/projectiles/guns/alien.dm]: Bay gave an ammo casing sharpness.

What.

File: code/modules/projectiles/guns/throw/crossbow.dm [hunk 6]
```diff
@@ -155,7 +155,6 @@
 	throwforce = 20
 	w_class = WEIGHT_CLASS_NORMAL
 	sharp = 1
-	edge = 0
```
Comment (Fox-McCloud) [inline code/modules/projectiles/guns/throw/crossbow.dm]: instance number 3 of sharp but no edge; clearly Bay didn't watch anime.

**Comment by Ralta** [comment]
I think I cut myself on this PR.

File: code/modules/mob/living/simple_animal/parrot.dm (hunk 4 above)
Comment (SamHPurp) [inline code/modules/mob/living/simple_animal/parrot.dm]: Poly buff confirmed

**Comment by Fox-McCloud** [comment]
Aghhhh. conflicts.

I actually anticipated this.

File: code/modules/projectiles/guns/projectile/bow.dm [hunk 7]
```diff
@@ -77,7 +77,6 @@
 	desc = "Stab, stab, stab."
 	icon_state = "arrow"
 	force = 10
-	sharp = 1
```
Comment (Fox-McCloud) [inline code/modules/projectiles/guns/projectile/bow.dm]: What's up with Bay and their sharp casings.

File: code/__HELPERS/unsorted.dm [hunk 8]
```diff
@@ -1202,16 +1202,11 @@ var/global/list/common_tools = list(
...
-	if(O.edge) return 1
-	return 0
-
-//Whether or not the given item counts as cutting with an edge in terms of removing limbs
-/proc/has_edge(obj/O as obj)
-	if(!O) return 0
-	if(O.edge) return 1
+/proc/is_sharp(obj/O)
+	if(!O)
+		return 0
```
Comment (SamHPurp) [inline code/__HELPERS/unsorted.dm]: formatting, should we not just use `TRUE` & `FALSE` for a simple boolean proc like this? eschew obfuscation and all that

**Comment by KasparoVy** [comment]
CRAWLING IN MY SKIN

**Comment by Spacemanspark** [comment]
Wake me up

File: code/modules/mob/living/carbon/human/species/species.dm [hunk 9]
```diff
@@ -532,7 +531,6 @@
 	attack_sound = 'sound/weapons/slice.ogg'
 	miss_sound = 'sound/weapons/slashmiss.ogg'
 	sharp = 1
```
Comment (shazbot194) [inline code/modules/mob/living/carbon/human/species/species.dm]: So wait, Unathi, Tajaran, Vulp, Vox, and Kidan can also dismember people with bare hands?

Comment (SamHPurp) [inline code/modules/mob/living/carbon/human/species/species.dm]: @shazbot194 they always could. I've always been particularly salty over a vulp beheading a changeling, stopping his regeneration.

**Pull request merged**